autodocodec/src/Autodocodec/Aeson/Decode.hs

```diff
@@ -111,16 +118,18 @@ parseJSONContextVia codec_ context_ =
       CommentCodec _ c -> go value c
       ReferenceCodec _ c -> go value c
       RequiredKeyCodec k c _ -> do
-        valueAtKey <- (value :: JSON.Object) JSON..: k
-        go valueAtKey c JSON.<?> Key k
+        valueAtKey <- (value :: JSON.Object) JSON..: Compat.toKey k
+        go valueAtKey c JSON.<?> Key (Compat.toKey k)
```

Comment:
I can't tell whether the `k` should be a `Key` or converted to a key here. What do you think?

Reply:
It's an interesting one. So far it seems like the Codec type has managed to avoid taking dependencies on Aeson types inside itself, other than the new KeyMapCodec added in this PR.

It feels like that's a good thing, because what if you wanted to write a codec interpreter that wasn't constrained by the same requirements as JSON? For example, (and this would be a change from what we have now) what if we wanted to say that a required key could be a numeric type? JSON doesn't allow that, but maybe some other format does?

Perhaps that's a good argument for keeping aeson types out of Codec, or perhaps I'm just gold plating for use cases that don't exist... thoughts?

Reply:
If we end up adopting KeyMapCodec as a thing, then I think we should bite the bullet and make `k` of type `Key` because we may as well go all in on the coupling to aeson at that point.